app/src/main/java/uk/gov/onelogin/tokens/usecases/GetFromOpenSecureStore.kt

```diff
@@ -0,0 +1,41 @@
+package uk.gov.onelogin.tokens.usecases
+
+import android.util.Log
+import javax.inject.Inject
+import javax.inject.Named
+import uk.gov.android.securestore.RetrievalEvent
+import uk.gov.android.securestore.SecureStore
+
+fun interface GetFromOpenSecureStore {
```

Comment:
Turning Types and Objects into functions by overriding the invoke operator is pointless as functions are already treated as first class objects and can be passed around freely. It also causes the heinous naming convention where suddenly you have Objects, Types, and properties that should be called nouns named as verbs, which lowers the reading comprehension. 

Coding should be as simple as humanly possible, and "clever" or extremely language idiomatic practices should be avoided and only used as a last resort. I do understand that this is done elsewhere, so I don't expect you to change it, but I will be advocating these instances be changed on mass, if not individually.

Always willing to hear other arguments though.

Reply:
We can discuss this as it's quite a large feature of our codebase